source/common/http/filter/ip_tagging_filter.h

```diff
@@ -5,52 +5,98 @@
 #include <string>
 #include <vector>
 
+#include "envoy/common/exception.h"
+#include "envoy/config/filter/http/ip_tagging/v2/ip_tagging.pb.h"
 #include "envoy/http/filter.h"
-#include "envoy/json/json_object.h"
+#include "envoy/runtime/runtime.h"
 
-#include "common/common/assert.h"
-#include "common/json/config_schemas.h"
-#include "common/json/json_validator.h"
+#include "common/network/cidr_range.h"
+#include "common/network/lc_trie.h"
 
 namespace Envoy {
 namespace Http {
 
 /**
  * Type of requests the filter should apply to.
  */
-enum class FilterRequestType { Internal, External, Both };
+enum class FilterRequestType { INTERNAL, EXTERNAL, BOTH };
 
 /**
- * Configuration for the ip tagging filter.
+ * Configuration for the HTTP IP Tagging filter.
  */
-class IpTaggingFilterConfig : Json::Validator {
+class IpTaggingFilterConfig {
 public:
-  IpTaggingFilterConfig(const Json::Object& json_config)
-      : Json::Validator(json_config, Json::Schema::IP_TAGGING_HTTP_FILTER_SCHEMA),
-        request_type_(stringToType(json_config.getString("request_type", "both"))) {}
+  IpTaggingFilterConfig(const envoy::config::filter::http::ip_tagging::v2::IPTagging& config,
+                        const std::string& stat_prefix, Stats::Scope& scope,
+                        Runtime::Loader& runtime)
+      : request_type_(requestTypeEnum(config.request_type())), scope_(scope), runtime_(runtime),
+        stats_prefix_(stat_prefix + "ip_tagging.") {
 
+    // Once loading IP tags from a file system is supported, the restriction on the size
+    // of the set should be removed and observability into what tags are loaded needs
+    // to be implemented.
+    // TODO(ccaraman): Remove size check once file system support is implemented.
```

Comment:
Is there a tracking issue for this feature?

Reply:
https://github.com/envoyproxy/envoy/issues/2695 - i'll change the todo to be the issue number.